portfolio/src/main/webapp/script.js

```diff
@@ -12,26 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Identity theft is not a joke', 'I declare bankruptcy', 'I am a little stitious'];
+getComments();
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+// submit a comment
+document.querySelector('.comment-form').addEventListener('submit', async function(e) {
+  e.preventDefault();
+  const response = await fetch('/comment', {
+    method: 'POST',
+    body: new URLSearchParams(new FormData(event.target)),
+  });
+  const json = await response.json();
+  if ('error' in json) {
+    console.log('Could not submit comment');
+  }
+  const commentContainer = document.querySelector('.comments');
+  commentContainer.innerHTML += '<div class="comment">' + document.querySelector('.comment-box').value + '</div><br>';
```

Comment:
If we get "" as the comment from user, what would document.querySelector('.comment-box').value be? Will we still print out ""?

Reply:
It will send "" in the request but in the servlet it checks for null or empty string on the sent parameter and won't store that comment and therefore that comment will never get sent to the frontend and printed out.